rust/src/nasl/utils/hosts.rs

```diff
@@ -2,18 +2,22 @@
 //
 // SPDX-License-Identifier: GPL-2.0-or-later WITH x11vnc-openssl-exception
 
-use std::net::{SocketAddr, ToSocketAddrs};
+use std::net::ToSocketAddrs;
 
 use crate::nasl::utils::error::FunctionErrorKind;
 
-pub fn resolve(
-    mut hostname: String,
-) -> Result<Option<Box<dyn Iterator<Item = SocketAddr>>>, FunctionErrorKind> {
+pub fn resolve(mut hostname: String) -> Result<Vec<String>, FunctionErrorKind> {
     //std::net to_socket_addrs() requires a port. Therefore, using a dummy port
     hostname.push_str(":5000");
 
     match hostname.to_socket_addrs() {
-        Ok(addr) => Ok(Some(Box::new(addr))),
+        Ok(addr) => {
+            let ips = addr
+                .into_iter()
+                .map(|x| x.ip().to_string())
```

Comment:
Why the `.to_string()`? It makes more sense to me to be explicit about what this function returns and to simply return the `IpAddr` instead.